src/sparseml/transformers/utils/sparse_model.py

```diff
@@ -13,35 +13,42 @@
 # limitations under the License.
 
 import inspect
+import json
 import logging
 import os
 from pathlib import Path
 from typing import Any, Dict, Optional, Tuple, Union
 
 import torch
 from torch.nn import Module
+from tqdm import tqdm
 from transformers import (
+    AutoConfig,
     AutoModelForCausalLM,
     AutoModelForMaskedLM,
     AutoModelForQuestionAnswering,
     AutoModelForSequenceClassification,
     AutoModelForTokenClassification,
+    PreTrainedModel,
 )
-from transformers.file_utils import WEIGHTS_NAME
+from transformers.file_utils import CONFIG_NAME, WEIGHTS_NAME
 
 from sparseml.pytorch.model_load.helpers import (
     apply_recipe_structure_to_model,
     log_model_load,
 )
+from sparseml.transformers.compression import CompressionConfig, ModelCompressor
 from sparseml.transformers.utils.helpers import resolve_recipe
 from sparseml.utils import download_zoo_training_dir
 from sparseml.utils.fsdp.context import main_process_first_context
 
 
-__all__ = ["SparseAutoModel", "SparseAutoModelForCausalLM", "get_shared_tokenizer_src"]
+_LOGGER = logging.getLogger(__name__)
 
+SPARSITY_CONFIG_NAME = "sparsity_config"
```

Comment:
potentially move to helpers/utils, this is where all the const reside for now

Reply:
done!